File: Python/2000.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        idx = word.find(ch)
        if idx == -1: return word
        return word[idx::-1] + word[idx+1:]

# Topic: String
# Time Complexity: O(n)
# Space Complexity: O(1)

File: C++/2000.cpp
#include <string>
#include <algorithm>

class Solution {
public:
    std::string reversePrefix(std::string word, char ch) {
        int idx = word.find(ch);
        if(idx != -1) std::reverse(word.begin(), word.begin()+idx+1);
        return word;
             
    }
};

// Topic: String
// Time Complexity: O(n)
// Space Complexity: O(1)

